### PR TITLE
[Improve][Connector-v2] Use regex to match filedName placeholders in jdbc sink

### PR DESCRIPTION
**Pull request by dailai**
### Purpose of this pull request

When the filed name of sink table containes '-' that can lead to this error
截屏2024-12-05 16 47 57

This is because the method named Character.isJavaIdentifierPart which is in FieldNamedPreparedStatement that  return false when the param is '-'.  So that the filedName will be split into two filedName.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Is there a way to support all special characters? e.g. "$","%","@" etc.

**Comment by dailai** [comment]
> Is there a way to support all special characters? e.g. "$","%","@" etc.



> Is there a way to support all special characters? e.g. "$","%","@" etc.

I have considered supporting all special characters, because the code in the current framework obtains relevant information by parsing the string sql, so it is costly to support all special characters.

Because this special character is not common at present, otherwise there would not be a case reported until now. Therefore, this change is the most efficient at present.

If there are more feedback in the future, consider more general support.

**Comment by dailai** [comment]
> Is there a way to support all special characters? e.g. "$","%","@" etc.

I use a regular expression to match as many characters as we can, but because the framework uses :filedName as a placeholder and computes index from filedName in that placeholder, there's still a problem if filedName also appears : character.


After this pr, we can support filedName placeholder like this:

:USER@TOKEN
:字段%名称
:field_name
:field.name
:field-name
:$fieldName
:field&key
:field*value
:field#1
:field~test
:field!data
:field?question
:field^caret
:field+add
:field=value
:field<min
:field>max
:field|pipe

**Comment by dailai** [comment]
Link pr https://github.com/apache/seatunnel/pull/7297

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/executor/FieldNamedPreparedStatement.java [hunk 1]
```diff
@@ -667,29 +669,25 @@ public static FieldNamedPreparedStatement prepareStatement(
...
-                paramMap.computeIfAbsent(parameterName, n -> new ArrayList<>()).add(fieldIndex);
-                fieldIndex++;
-                i = j - 1;
-                parsedSql.append('?');
-            } else {
-                parsedSql.append(c);
-            }
+    public static String parseNamedStatement(String sql, Map<String, List<Integer>> paramMap) {
+        Pattern pattern =
+                Pattern.compile(":([\\p{L}\\p{Nl}\\p{Nd}\\p{Pc}\\$\\-\\.@%&*#~!?^+=<>|]+)");
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/executor/FieldNamedPreparedStatement.java:679]: Please update the test case to verify the new type char.

Comment (dailai) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/executor/FieldNamedPreparedStatement.java:679]: done

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/executor/FieldNamedPreparedStatement.java [hunk 2]
```diff
@@ -669,29 +671,25 @@ public static FieldNamedPreparedStatement prepareStatement(
...
-                        !parameterName.isEmpty(),
-                        "Named parameters in SQL statement must not be empty.");
-                paramMap.computeIfAbsent(parameterName, n -> new ArrayList<>()).add(fieldIndex);
-                fieldIndex++;
-                i = j - 1;
-                parsedSql.append('?');
-            } else {
-                parsedSql.append(c);
-            }
+    public static String parseNamedStatement(String sql, Map<String, List<Integer>> paramMap) {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/executor/FieldNamedPreparedStatement.java:677]:
```suggestion
    @VisibleForTesting
    public static String parseNamedStatement(String sql, Map<String, List<Integer>> paramMap) {
```

**Review by Hisoka-X — APPROVED** [review]
LGTM if ci passes.

**Review by hailin0 — APPROVED** [review]

**Comment by dailai** [comment]
Let me merge it. The ci is no problem. The failed ci link pr #8262 .

**Pull request merged**